metrics/prometheus/preload.go

```diff
@@ -207,6 +208,40 @@ func preloadLabelValues(m *Metrics, syncerKeys []string) {
 			adapterLabel: adapterValues,
 		})
 	}
+
+	for module := range moduleStageNames {
+		preloadLabelValuesForHistogram(m.moduleDuration[module], map[string][]string{
+			stageLabel: stageValues,
```

Comment:
Shouldn't this look more like:

```
for module, stageValues := range moduleStageNames {
  preloadLabelValuesForHistogram(m.moduleDuration[module], map[string][]string{
    stageLabel: stageValues,
```

This way we only pre-create labels which are needed for the particular module.

.. or else just use all possible stage values instead.

Reply:
I agree with you. I made that all possible stages are added for every module, but that's unnecessary. Fixed